src/com/google/javascript/jscomp/CheckJSDoc.java

```diff
@@ -0,0 +1,274 @@
+      if (info == null) {
+        return;
+      }
+      if (info.containsFunctionDeclaration() && !info.hasType()) {
+        // This JSDoc should be attached to a FUNCTION node, or an assignment
+        // with a function as the RHS, etc.
+        switch (n.getType()) {
+          case Token.FUNCTION:
+          case Token.VAR:
+          case Token.LET:
+          case Token.CONST:
+          case Token.GETTER_DEF:
+          case Token.SETTER_DEF:
+          case Token.MEMBER_FUNCTION_DEF:
+          case Token.STRING_KEY:
+          case Token.EXPORT:
+            return;
+          case Token.GETELEM:
+          case Token.GETPROP:
+            if (n.getFirstChild().isQualifiedName()) {
+              return;
+            }
+            break;
+          case Token.ASSIGN: {
+            // TODO(tbreisacher): Check that the RHS of the assignment is a
+            // function. Note that it can be a FUNCTION node, but it can also be
+            // a call to goog.abstractMethod, goog.functions.constant, etc.
+            return;
+          }
+        }
+        t.getCompiler().report(t.makeError(n, MISPLACED_ANNOTATION,
+            "function", "This JSDoc is not attached to a function node. "
+                + "Are you missing parentheses?"));
+      }
+    }
+
+    /**
+     * Checks that annotations for messages ({@code @desc}, {@code @hidden},
+     * and {@code @meaning})
+     * are in the proper place, namely on names starting with MSG_ which
+     * indicates they should be
+     * extracted for translation. A later pass checks that the right side is
+     * a call to goog.getMsg.
+     */
+    private void validateMsgJsDoc(NodeTraversal t, Node n) {
+      JSDocInfo info = n.getJSDocInfo();
+      if (info == null) {
+        return;
+      }
+      if (info.getDescription() != null || info.isHidden() || info.getMeaning() != null) {
+        boolean descOkay = false;
+        switch (n.getType()) {
+          case Token.ASSIGN: {
+            Node lhs = n.getFirstChild();
+            if (lhs.isName()) {
+              descOkay = lhs.getString().startsWith("MSG_");
+            } else if (lhs.isQualifiedName()) {
+              descOkay = lhs.getLastChild().getString().startsWith("MSG_");
+            }
+            break;
+          }
+          case Token.VAR:
+          case Token.LET:
+          case Token.CONST:
+            descOkay = n.getFirstChild().getString().startsWith("MSG_");
+            break;
+          case Token.STRING_KEY:
+            descOkay = n.getString().startsWith("MSG_");
+            break;
+        }
+        if (!descOkay) {
+          t.getCompiler().report(t.makeError(n, MISPLACED_ANNOTATION,
+              "message", " @desc, @hidden, and @meaning annotations should only"
+                  + "be on message nodes."));
+        }
+      }
+    }
+
+    /**
+     * Check that JSDoc with a {@code @type} annotation is in a valid place.
+     */
+    @SuppressWarnings("incomplete-switch")
```

Comment:
it seems that you can just add a default: case for the switch and remove this suppression